src/runtime/methodbinder.cs

```diff
@@ -507,7 +507,7 @@ internal virtual IntPtr Invoke(IntPtr inst, IntPtr args, IntPtr kw, MethodBase i
 
             if (binding == null)
             {
-                Exceptions.SetError(Exceptions.TypeError, "No method matches given arguments");
+                Exceptions.SetError(Exceptions.TypeError, $"No method matches given arguments for {methodinfo[0].Name}");
```

Comment:
How do you know that `methodinfo` is not `null` and that it contains at least one element?

Reply:
Right, I will add a check for null and at least one element.
Thank you. I didn't came across those conditions in my application.